funding/manager.go

```diff
@@ -152,13 +152,32 @@ func (m *Manager) deriveFundingShim(ourOrder order.Order,
 
 		thawHeight      uint32
 		selfChanBalance btcutil.Amount
+		ourMultiSigKey  *keychain.KeyDescriptor
 	)
-	if ourOrder.Type() == order.TypeBid {
+	ourOrderBid, ourOrderIsBid := ourOrder.(*order.Bid)
+	ourOrderIsSidecar := ourOrderIsBid &&
+		ourOrderBid.SidecarTicket != nil
+
+	if ourOrderIsBid {
 		bidNonce = ourOrder.Nonce()
 		askNonce = matchedOrder.Order.Nonce()
 
-		thawHeight = ourOrder.(*order.Bid).LeaseDuration
-		selfChanBalance = ourOrder.(*order.Bid).SelfChanBalance
+		thawHeight = ourOrderBid.LeaseDuration
+		selfChanBalance = ourOrderBid.SelfChanBalance
+
+		// If our order is a sidecar channel, we already have the
+		// multisig key in the ticket and don't need to re-derive it
+		// again here.
+		ticket := ourOrderBid.SidecarTicket
+		if ourOrderIsSidecar && ticket.Recipient != nil {
```

Comment:
can we move this down to an `else` clause with the other option instead?

Reply:
So you mean turn this into a `switch` statement effectively?

Reply:
Extracted just this `if` to make it a bit more readable. A switch wouldn't be optimal as we'd have some duplication.